### ISSUE TITLE
Allow rover processes to subscribe to message after startup.

### ISSUE THREAD

**Issue by ottopasuuna** [UofSSpaceDesignTeam/roveberrypy#36]
I'm currently running into an issue in the 2016-refactoring branch where a rover process can call the addSubscriber method on the StateManager to subscribe to messages durring runtime. Unfortunately, although the StateManager instance's subscriberMap is updated, the WorkerThread does not get the new entry, and thus messages are not pushed to the process.

**Comment by ottopasuuna** [comment]
I found out that the issue is related to shared data accross multiple processes. We have been using multiprocessing.Queue because those data structures are designed to work across processes. Regular dictionaries do not. When the process begins, it gets a copy of it's parent's memory, and cannot tell if values have been updated in another process. One way around this is multiprocessing.Manager, which provides a proxy server of sorts to ensure that values are updated between threads. Unfortunately, python can't handle a dictionary of lists of Queues due to a limitation (?) of pickle, the serialization library python uses in multiprocessing.Manager.

So in otherwords, implementing this feature would be best done as part of a complete rewrite of the rover software, something that will have to wait for Chrismas break (or later).
